### PR TITLE
completed most of lab

### PR DESCRIPTION
**Pull request by tcoots88**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: code401challenges/src/main/java/code401challenges/linkedList/LinkedList.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package code401challenges.linkedList;
+
+import java.util.*;
+
+
+public class LinkedList {
+    Node head;
+
+    public void insert(int value) {
```
Comment (JamesDansie) [inline code401challenges/src/main/java/code401challenges/linkedList/LinkedList.java]: This inserts at the tail. The assignment was to insert at the head. Don't delete it, you'll need it later.

File: code401challenges/src/main/java/code401challenges/linkedList/LinkedList.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        Node node = new Node();
+        node.data = value;
+        node.next = null;
+
```
Comment (JamesDansie) [inline code401challenges/src/main/java/code401challenges/linkedList/LinkedList.java]: When targeting head you want to have this.head, otherwise it can get confused with other head variables.

File: code401challenges/src/main/java/code401challenges/linkedList/LinkedList.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            Node n = head;
+            while (n.next != null) {
+                n = n.next;
+            }
+            n.next = node;
+        }
+    }
+    public void show() {
+        Node node = head;
+        while(node.next != null) {
```
Comment (JamesDansie) [inline code401challenges/src/main/java/code401challenges/linkedList/LinkedList.java]: Instead of checking node.next check node. This way you don't have to manually add the last node at the end. So use;
while(node != null)

File: code401challenges/src/main/java/code401challenges/linkedList/LinkedList.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                nextNode = nextNode.next;
+        }
+        return false;
+    }
+
+    public String toString() {
+        String values = "";
+
+        for(Node curr = head; curr != null; curr = curr.next) {
+            values = values + curr.data;
```
Comment (JamesDansie) [inline code401challenges/src/main/java/code401challenges/linkedList/LinkedList.java]: You'll want something to separate the node values (a comma or {} would be nice)
```suggestion
            values = values + ", " + curr.data;
```

File: code401challenges/src/test/java/code401challenges/linkedList/LinkedListTest.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        LinkedList newLinkedList = new LinkedList();
+        assertNull("Empty List was not made successfully, RIP", newLinkedList.head);
+    }
+
+    @Test
+    public void testIncludesMethodTrue() {
+        LinkedList newLinkedList = new LinkedList();
+        newLinkedList.insert(15);
+
+        assertTrue("Linked List did not include target value", newLinkedList.includes(15));
```
Comment (JamesDansie) [inline code401challenges/src/test/java/code401challenges/linkedList/LinkedListTest.java]:
```suggestion
        assertTrue("Linked List did include target value", newLinkedList.includes(15));
```
